error_post_go120.go

```diff
@@ -27,3 +27,22 @@ package multierr
 func (merr *multiError) Unwrap() []error {
 	return merr.Errors()
 }
+
+type multipleErrors interface {
+	Unwrap() []error
+}
+
+func extractErrors(err error) []error {
+	if err == nil {
+		return nil
+	}
+
+	// check if the given err is an Unwrapable error that
+	// implements multipleErrors interface.
+	eg, ok := err.(multipleErrors)
+	if !ok {
+		return []error{err}
+	}
+
+	return append(([]error)(nil), eg.Unwrap()...)
```

Comment:
out of curiosity, why a new slice rather than the old one?

it would seem that any unrwappable error should be okay with us passing the slice up?

Reply:
If memory serves, it's a new slice because multierr.Errors specifically returns a copy of the slice.
Docs say:

> Callers of this function are free to modify the returned slice.

`Unwrap() []error` returns the underlying slice. Per [the proposal](https://github.com/golang/go/issues/53435#issuecomment-1191752789):

> Callers must not modify the list returned by Unwrap.

And indeed, in the [implementation](https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/errors/join.go;l=49-51):

```go
func (e *joinError) Unwrap() []error {
	return e.errs
}
```

So we have to create a new slice here.